packages/inventory/src/Stocktake/DetailView/DetailView.tsx

```diff
@@ -15,21 +15,29 @@ import { Toolbar } from './Toolbar';
 import { Footer } from './Footer';
 import { AppBarButtons } from './AppBarButtons';
 import { SidePanel } from './SidePanel';
-import { StocktakeLine, StocktakeSummaryItem } from '../../types';
+import { StocktakeSummaryItem } from '../../types';
 
 import { StocktakeLineEdit } from './modal/StocktakeLineEdit';
 import { ContentArea } from './ContentArea';
 import { useStocktake } from '../api/hooks';
 import { AppRoute } from '@openmsupply-client/config';
+import { StocktakeLineFragment } from '../api';
 
 export const DetailView: FC = () => {
   const { isOpen, entity, onOpen, onClose, mode } = useEditModal<Item>();
   const { data, isLoading } = useStocktake();
   const navigate = useNavigate();
   const t = useTranslation('inventory');
 
-  const onRowClick = (item: StocktakeLine | StocktakeSummaryItem) => {
-    onOpen(toItem(item));
+  const onRowClick = (item: StocktakeLineFragment | StocktakeSummaryItem) => {
+    if ('lines' in item) return onOpen(toItem(item));
+    onOpen(
+      toItem({
+        itemId: item.item?.id ?? '',
+        itemName: item.item?.name ?? '',
+        itemCode: item.item?.code ?? '',
+      })
+    );
```

Comment:
This is a little annoying and will need to revisit how this works properly rather than the hack here. Essentially the `toItem` was using a `ItemLike` parameter which was a `line` that had properties `itemId`, `itemName`, `itemCode` and turning it into an `Item` type - now that `StocktakeLine` doesn't have those properties, rather just the `item` field, it's a little different. 

`RequisitionLineNode` and `StocktakeLineNode` don't have those properties, but `InvoiceLineNode` does - all I want is consistency so I can use the same function for them all! 

Thinking we should just go with using an `item` field - though, for invoices, that means using the guard and things as the invoice types are using the 'older' pattern

Reply:
sounds good!